src/hooks/useStickyIntersectionObserver.js

```diff
@@ -0,0 +1,55 @@
+import {
+  useEffect
+} from 'preact/hooks';
+
+import {
+  query as domQuery
+} from 'min-dom';
+
+
+/**
+ * @callback setSticky
+ * @param {boolean} value
+ */
+
+/**
+ * Use IntersectionObserver to identify when DOM element is in sticky mode.
+ * If sticky is observered setSticky(true) will be called.
+ * If sticky mode is left, setSticky(false) will be called.
+ *
+ *
+ * @param {Object} ref
+ * @param {setSticky} setSticky
+ */
+export function useStickyIntersectionObserver(ref, scrollContainerSelector, setSticky) {
+  useEffect(() => {
+    let observer;
+
+    if (ref.current) {
+      const scrollContainer = domQuery(scrollContainerSelector);
+
+      observer = new IntersectionObserver((entries) => {
```

Comment:
IE has[ no support](https://caniuse.com/?search=IntersectionObserver) for this, but I think we are fine with that, right?

Reply:
Yes, that would be my understanding. 

@nikku and @barmac also discussed with me that we can use `position: sticky`, which is also not supported by IE.